src/interfaces/wallet.cpp

```diff
@@ -370,7 +371,11 @@ class WalletImpl : public Wallet
         balances = getBalances();
         return true;
     }
-    CAmount getBalance() override { return m_wallet->GetBalance().m_mine_trusted; }
+    CAmount getBalance() override
+    {
+        LOCK(m_wallet->cs_wallet);
+        return m_wallet->GetBalance().m_mine_trusted;
+    }
```

Comment:
Probably, it would be a cleaner code with two methods:
- `getBalance() EXCLUSIVE_LOCKS_REQUIRED(!m_wallet->cs_wallet)`
- `getBalance_cs() EXCLUSIVE_LOCKS_REQUIRED(m_wallet->cs_wallet)`

See:
- https://github.com/bitcoin/bitcoin/pull/19238#issuecomment-646226615
- a7b1a649ec473bd9811e34cf489bd88b3c3d6406 (#19238)

Reply:
It could replace 6 instances of added `WITH_LOCK()` :)

Reply:
you mean `getBalance` or `GetBalance`? I don't see why we want that on `interfaces::Wallet`. If you mean `GetBalance` then ATM I tend to prefer explicit `WITH_LOCK`/`LOCK` for so few cases. If we go that way then I'd prefer to overload `GetBalance(Lock)`.

Reply:
> Do you mean `getBalance` or `GetBalance`?

My suggestion is wrong and confusing. Please ignore it.

> If we go that way then I'd prefer to overload `GetBalance(Lock)`.

I'm afraid Clang Thread Safety Analysis won't understand it due to https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#no-alias-analysis